src/python/WMCore/WMSpec/Steps/Executor.py

```diff
@@ -51,8 +53,39 @@ def getStepSpace(stepName):
         raise RuntimeError(msg)
     return stepSpace
 
+def zipEncodeStr(message, maxLen=5120, compressLevel=9, steps=100, truncateIndicator=" (...)"):
```

Comment:
Given this function doesn't depend on anything WMCore related, how about we move it under 
https://github.com/dmwm/WMCore/blob/master/src/python/Utils/Utilities.py
?

A unit test testing different scenarios is also very important in this case.